### PR TITLE
Exception tests

### PR DESCRIPTION
**Pull request by darianbr**
First pass at peridot.

### CODE REVIEW

File: test/suite/unit/Exception/FrozenAttributeException.spec.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+declare(strict_types=1); // @codeCoverageIgnore
+
+namespace Rinq\Exception;
+
+use Rinq\Ident\PeerId;
+use Rinq\Ident\Reference;
+use Rinq\Ident\SessionId;
+
+describe('FrozenAttributeException', function () {
```
Comment (jmalloc) [inline test/suite/unit/Exception/FrozenAttributeException.spec.php]: I like to do `FrozenAttributeException::class` for the outer description. Your call, it just helps when you end up with the same class name used in different namespaces.

File: test/suite/unit/Exception/StaleUpdateException.spec.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    beforeEach(function () {
+        $this->reference = Reference::create(SessionId::create(PeerId::create(111, 222), 333), 444);
+        $this->subject = new StaleUpdateException($this->reference);
+    });
+
+    describe('->construct()', function () {
+        it('should construct a meaningful exception message', function () {
+            expect($this->subject->getMessage())->to->equal(
+                'Can not update or close 6F-00DE.333@444, the session has ' .
+                'been modified since that revision.'
```
Comment (jmalloc) [inline test/suite/unit/Exception/StaleUpdateException.spec.php]: This is *really* picky, but if you're gonna split strings, it should probably be tabbed in on the second line to make it clear that it's part of the same expression.  My personal preference is to simply not split them, but it's your call.

**Review by jmalloc — APPROVED** [review]
A couple of comments that are really personal preference, but happy for this to go in as-is.

File: test/suite/unit/Exception/FrozenAttributeException.spec.php (hunk 1 above)
Comment (darianbr) [reply test/suite/unit/Exception/FrozenAttributeException.spec.php]: Good call, and I like the way it reads.

File: test/suite/unit/Exception/StaleUpdateException.spec.php (hunk 2 above)
Comment (darianbr) [reply test/suite/unit/Exception/StaleUpdateException.spec.php]: I generally only split the strings that exceed 120 chars. I have indented the second line in those cases.

**Pull request merged**